### ISSUE TITLE
Describing the difference between HyperGraphQL and OSLC Query

### ISSUE THREAD

**Issue by jamsden** [jamsden/oslc-core-test#17]
HyperGraphQL ([http://hypergraphql.org/](http://hypergraphql.org/) ) looks very similar in functionality to OSLC Query. HyperGraphQL seems to have reached a relevant level of significance in the broad Linked Data community, as it is based on the popular GraphQL standard. At some point, developers will want to know the difference between HyperGraphQL and OSLC Query. Understanding these differences may also shape future OSLC Query standardization efforts.

**Comment by berezovskyi** [comment]
I also agree with David & Jim that the TC‘s focus for Query 3.0 would be to maintain compat with 2.0 as well as to cater to the implementations in the wild. However, I don‘t see anything in Axel‘s proposal to modify the normative parts of the Query 3.0 spec.

**Comment by berezovskyi** [comment]
I think the most interesting thing for OSLC in re other approaches to querying and sharing updates would be to support the discovery of those mechanisms (incl. SPARQL endpoints and other mentioned mechanisms as well as LDN in addition to TRS etc.).

**Comment by DavidJHoney** [comment]
OLSC 3.0 Query needs to be compatible with the 2.0 Query. There are other competing ways that tools might provide querying/reporting capabilities including Reportable REST API, SPARQL, and Gremlin. I think we should avoid offering too many overlapping solutions as it adds to the implementation burden and causes confusion among adopters.The main motivation for OSLC Query was to provide a much-easier-to-use query capability that was persistence agnostic. It only partly succeeded because Query 2.0 required too strict conformance with SPARQL that burdened implementations to persist data as RDF to use SPARQL against.

Many customers will want cross-cutting querying, and for this, customers using Indexing providers such as LQE may find that SPARQL based on Jena does not provide the scalability they need for the future. Graph-based databases and querying seems a promising solution to this. This is a rapidly evolving area.

I think we should wait and see what customer use cases are not adequately supported by OSLC Query or SPARQL before choosing one of several candidate graph-based query languages.
